### PR TITLE
feat: configurable cert refresh time for kubeconfig

### PR DESCRIPTION
**Pull request by frezbo**
Support configurable cert refresh time for `talos_cluster_kubeconfig`.

Fixes: #203

### CODE REVIEW

**Comment by frezbo** [comment]
@hegerdes would this work for you, not sure if the naming is perfect

**Comment by hegerdes** [comment]
That would be an awesome feature!

Thanks for the fast actions.

File: pkg/talos/talos_cluster_kubeconfig_resource.go [hunk 1]
```diff
@@ -451,9 +636,16 @@ func (r *talosClusterKubeConfigResource) Update(ctx context.Context, req resourc
...
 	}
 
+	renewalDuration, err := time.ParseDuration(state.CertificateRenewalDuration.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("failed to parse certificate renewal duration", err.Error())
+
+		return
+	}
+
 	// check if NotAfter expires in a month
```
Comment (smira) [inline pkg/talos/talos_cluster_kubeconfig_resource.go:646]: probably comment is now outdated?

**Review by smira — APPROVED** [review]

File: pkg/talos/talos_cluster_kubeconfig_resource.go (hunk 1 above)
Comment (frezbo) [reply pkg/talos/talos_cluster_kubeconfig_resource.go:646]: ah yyess

**Pull request merged**